x-pack/libbeat/management/managerV2.go

```diff
@@ -378,3 +393,44 @@ func (cm *BeatV2Manager) handleInputReload(unit *client.Unit) {
 	}
 	_ = unit.UpdateState(client.UnitStateHealthy, "beat reloaded", nil)
 }
+
+// this function is registered as a debug hook
+// it prints the last known configuration genreated by the beat
+func (cm *BeatV2Manager) handleDebugYaml() []byte {
+	// generate input
+	inputList := []map[string]interface{}{}
+	for _, module := range cm.lastInputCfg {
+		var inputMap map[string]interface{}
+		err := module.Config.Unpack(&inputMap)
+		if err != nil {
+			cm.logger.Errorf("error unpacking input config for debug callback: %s", err)
+			return nil
+		}
+		inputList = append(inputList, inputMap)
+	}
+
+	// generate output
+	outputCfg := map[string]interface{}{}
+	err := cm.lastOutputCfg.Config.Unpack(&outputCfg)
+	if err != nil {
+		cm.logger.Errorf("error unpacking output config for debug callback: %s", err)
+		return nil
+	}
+	// combine the two in a somewhat coherent way
+	// This isn't perfect, but generating a config that can actually be fed back into the beat
+	// would require
```

Comment:
```suggestion
	// would require
```

We appear to be missing the rest of this sentence.

Reply:
Yah, looks like something I would do...